File: Xamarin/MetronomeApp/MetronomeApp/MainActivity.cs
using System;
using System.Diagnostics;

using Android.App;
using Android.Media;
using Android.Content;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using Android.OS;

namespace MetronomeApp
{
	[Activity (Label = "MetronomeApp", MainLauncher = true, Icon = "@drawable/icon")]
	public class MainActivity : Activity
	{
		MediaPlayer _player;
		int BPM;
		long BeforeTime;
		Stopwatch StopWatch = new Stopwatch();

		protected override void OnCreate (Bundle bundle)
		{
			base.OnCreate (bundle);

			_player = MediaPlayer.Create(this, MetronomeApp.Resource.Raw.Metronome1a);

			// Set our view from the "main" layout resource
			SetContentView (MetronomeApp.Resource.Layout.Main);
			//Get ui elements
			Android.Widget.Switch StartSwitch = FindViewById<Android.Widget.Switch>(MetronomeApp.Resource.Id.Switch);
			EditText BPMText = FindViewById<EditText>(MetronomeApp.Resource.Id.BPM);

			StartSwitch.CheckedChange += delegate(object sender, CompoundButton.CheckedChangeEventArgs e) {
				BPM = Convert.ToInt16 (BPMText.Text);

				StopWatch.Start ();
				WatchStart:
				_player.Start ();
				BeforeTime = StopWatch.ElapsedMilliseconds;
				IfStart:
				if (BeforeTime + 60000 / BPM == StopWatch.ElapsedMilliseconds)
					goto WatchStart;
				else
					goto IfStart;
			};
		}
	}
}

File: AndroidStudio/MetronomeApp/app/src/main/java/com/petermorits/moritswille/metronomeapp/MainActivity.java
package com.petermorits.moritswille.metronomeapp;

import android.os.Bundle;
import android.os.CountDownTimer;
import android.support.design.widget.FloatingActionButton;
import android.support.design.widget.Snackbar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.EditText;
import android.widget.TextView;
import android.media.MediaPlayer;

public class MainActivity extends AppCompatActivity {
    MediaPlayer click = MediaPlayer.create(this, R.raw.metronome1a);
    boolean playOrNot = false;
    int bpm;


    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);
        fab.setOnClickListener(new View.OnClickListener() {
            public void onClick(View view) {
                FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);
                TextView text1 = (TextView) findViewById(R.id.text1);
                EditText bpmText = (EditText) findViewById(R.id.bpmText);
                bpm = Integer.parseInt(bpmText.getText().toString());
                if (playOrNot == true) {
                    text1.setText("Stop..");
                    playOrNot = false;
                } else {
                    metroPlay();
                    text1.setText("GO!");
                }
            }
        });
    }
    void metroPlay()
    {
        new CountDownTimer(bpm / 60000, 1000) {

            public void onTick(long millisUntilFinished) {

            }

            public void onFinish() {
                click.start();
            }
        }.start();
    }
    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }
}
